packages/fxa-settings/src/lib/auth-errors/auth-errors.ts

```diff
@@ -604,13 +620,26 @@ const ERRORS = {
 };
 
 type ErrorKey = keyof typeof ERRORS;
-type ErrorVal = { errno: number; message: string };
+type ErrorVal = { errno: number; message: string; version?: number };
+
+export const composeAuthUiErrorTranslationID = (errno?: number) => {
+  /* all of these checks for fields/values being present look a little wonky, but allow us to work with
+   * the AuthUiError type, which has all optional fields. Previously this wasn't an issue bc we didn't use
+   * a utility.
+   */
+  if (errno && errno in AuthUiErrorNos) {
+    const error = AuthUiErrorNos[errno];
+    return `auth-error-${errno}${error.version && ('-' + error.version)}`;
+  }
+  return '';
```

Comment:
Correct me if I'm wrong but I think we just want to throw an error here instead right? I don't know when we'd ever _want_ an empty string returned as the FTL ID.

Also what do you think about passing in the entire `error` object for one less lookup (avoids `const error = AuthUiErrorNos[errno]`)? Not a big deal though since that list is small. Can we add a couple tests for this new utility function though? 🙂

Reply:
Tests are a great idea! So I passed in just the error number bc I was stumbling over a lot of issues with the flexibility of the AuthUIError object -- sometimes we're passing in a specific AuthUIError object, sometimes we're passing in the error coming in from the catch/try -- and I was stumbling over some unexpectedly breaking tests and (expectedly, haha) occurring typescript errors. With those resolved now, I'll try throwing an error and see if that still passes tests!